server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -984,6 +984,21 @@ public boolean canUseSyntheticSourceDelegateForQuerying() {
                 && syntheticSourceDelegate.isIndexed();
         }
 
+        /**
+         * Returns true if the delegate sub-field can be used for querying only (ie. isIndexed must be true)
+         */
+        public boolean canUseSyntheticSourceDelegateForQueryingEquality(String str) {
+            if (syntheticSourceDelegate == null
+                // Can't push equality to an index if there isn't an index
+                || syntheticSourceDelegate.isIndexed() == false
+                // ESQL needs docs values to push equality
+                || syntheticSourceDelegate.hasDocValues() == false) {
+                return false;
+            }
+            // Can't push equality if the field we're checking for is so big we'd ignore it.
+            return str.length() < syntheticSourceDelegate.ignoreAbove();
```

Comment:
Could this be `<=` or is `ignore_above` inclusive?

Reply:
NOCOMMIT: Nik write a test for this boundary condition. Off by one strikes again?

Reply:
Yeah, was off by one. Pushed a fix.